omod/src/main/java/org/openmrs/module/mksreports/web/controller/MKSReportsManageController.java

```diff
@@ -13,196 +13,75 @@
 import org.openmrs.module.mksreports.MKSReportsConstants;
-import org.openmrs.module.mksreports.reports.PatientHistoryReportManager;
-import org.openmrs.module.patientsummary.PatientSummaryResult;
-import org.openmrs.module.patientsummary.PatientSummaryTemplate;
-import org.openmrs.module.patientsummary.api.PatientSummaryService;
-import org.openmrs.module.reporting.evaluation.context.EncounterEvaluationContext;
-import org.openmrs.module.reporting.query.encounter.EncounterIdSet;
-import org.openmrs.module.reporting.report.ReportDesign;
-import org.openmrs.module.reporting.report.service.ReportService;
-import org.openmrs.util.OpenmrsClassLoader;
+import org.openmrs.module.mksreports.reports.PatientHistoryPdfReport;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 
-/**
- * The main controller.
- */
 @Controller
 public class MKSReportsManageController {
 	
-	@Autowired
-	private EncounterService encounterService;
+	private PatientHistoryPdfReport pdfReport;
 	
-	/**
-	 * The path to the style sheet for Patient History reports.
-	 */
-	private static final String PATIENT_HISTORY_XSL_PATH = "patientHistoryFopStylesheet.xsl";
-	
-	@Autowired
-	private PatientSummaryService patientSummaryService;
+	private EncounterService es;
 	
 	@Autowired
-	private ReportService reportService;
+	public MKSReportsManageController(@Qualifier("encounterService") EncounterService es,
+	    PatientHistoryPdfReport pdfReport) {
+		this.es = es;
+		this.pdfReport = pdfReport;
+	}
 	
 	/**
 	 * Receives requests to run a patient summary.
 	 * 
 	 * @param patientId the id of patient whose summary you wish to view
-	 * @param summaryId the id of the patientsummary you wish to view
-	 * @param encounterId the id(s) of the encounters you wish to view, multiple encounters should be in
-	 *            the form "<uuid>,<uuid>"
+	 * @param encounterUuids the UUID(s) of the encounters you wish to view, CSV format is supported for
+	 *            multiple encounters
 	 * @param target if used as an href URL, the intended target of the <a> (e.g. _self, _blank)
 	 * @throws Exception
 	 */
 	@RequestMapping(value = MKSReportsConstants.CONTROLLER_PATIENTHISTORY_ROUTE)
 	public void renderPatientHistory(ModelMap model, HttpServletRequest request, HttpServletResponse response,
-	        @RequestParam("patientId") Integer patientId,
-	        @RequestParam(value = "encounterUuid", required = false) String encounterUuidParam,
+	        @RequestParam(value = "patientId", required = false) Integer patientId,
+	        @RequestParam(value = "encounterUuid", required = false) String encounterUuids,
 	        @RequestParam(value = "target", required = false) String target) {
```

Comment:
This should be named `contentDisposition`, let's leave it to the caller to specify whether it's `inline`, or `attachment`, or whatever the caller needs it to be. `target` doesn't mean anything and is typically a DOM term.

The JavaDocs also need to be updated.

Reply:
I don't have a better idea at this point than having this system of params:

1. If `patientUuid` is present then we proceed to downloading the whole patient history.
2. Else if `encounterUuid` is present then we proceed to downloading that encounter.
3. Else if a (new) `extraEncounterUuids` is present then we proceed to also download those on top of the one specified by `encounterUuid`.

So either we point to a patient or to his/her encounter(s).

Reply:
First, ok. I can see `contentDisposition` inline and attachment making sense. I don't know about adding anything else there though.

Even though there is a different meaning to 'attachment' in the contentDisposition context, your comment also had me thinking about the other case. Should I move the logic for creating a pdf complex obs "attachment" from the current implementer module to MKSReports? Should that be a different endpoint, or another parameter?

In response to your second comment, I had already added logic for these based on our conversation here:

https://github.com/kml27/openmrs-module-mksreports/blob/f0a6773df99b9323816bb3fc9b3068076716a047/api/src/main/java/org/openmrs/module/mksreports/reports/PatientHistoryPdfReport.java#L86-L113

If you feel we should break out the "additional" encounterUuids into `encounterUuid` and `extraEncounterUuids`, thereby requiring any request logic break up the set that way, it's your call, but we wound up doing this all server side instead of generating a link to this controller. I don't know of any place we're using the multiple encounter support *on this controller, at this time*.

Reply:
Also, somehow we lost the randomized filename in this function which was to replace the default `patientHistory.pdf`.